src/buffer.rs

```diff
@@ -169,25 +158,32 @@ impl<'source, T: Element> FromPyObject<'source> for PyBuffer<T> {
 impl<T: Element> PyBuffer<T> {
     /// Get the underlying buffer from the specified python object.
     pub fn get(obj: &PyAny) -> PyResult<PyBuffer<T>> {
-        unsafe {
-            let mut buf = Box::pin(ffi::Py_buffer::new());
+        // TODO: use nightly API Box::new_uninit() once stable
+        let mut buf = Box::new(mem::MaybeUninit::uninit());
```

Comment:
If you take this approach, then you can remove `Py_buffer::new()`.
Though I'm a bit doubtful about how effective `MaybeUninit` is (I mean that this is effective only when we raise an error, but then `buf` is immediately dropped...).

Reply:
In the future I would like to remove `Py_buffer::new()`, but I think to do so right now would be inconvenient for users because [`PyBufferProtocol`](https://docs.rs/pyo3/0.13.2/pyo3/class/buffer/trait.PyBufferProtocol.html) basically forces you to handle the raw ffi pointer (which probably points to uninitialized memory which C created).

I chose `MaybeUninit` here because that's pretty much identical to what C would do: we create a slab of memory and pass its location to `PyObject_GetBuffer` to initialize the contents.